edge/pkg/edged/volume/csi/csi_plugin.go

```diff
@@ -415,6 +415,8 @@ func (p *csiPlugin) NewMounter(
 	}
 	klog.V(4).Info(log("created path successfully [%s]", dataDir))
 
+	mounter.MetricsProvider = NewMetricsCsi(volumeHandle, dir, csiDriverName(driverName))
```

Comment:
Is it possible to import from k8s instead of copy in? :)

Reply:
@fisherxu woud that be OK for you to copy/paste that part?

Reply:
The `csiDriverName` is not included in `[volume/csi/csi_metrics.go]`, so I just tested in my host:

in [csi_plugin.go]:

```
+       "k8s.io/kubernetes/pkg/volume/csi"
 
        "github.com/kubeedge/kubeedge/edge/pkg/edged/volume/csi/nodeinfomanager"
 )
@@ -415,6 +416,9 @@ func (p *csiPlugin) NewMounter(
        }
        klog.V(4).Info(log("created path successfully [%s]", dataDir))
 
+       mounter.MetricsProvider = csi.NewMetricsCsi(volumeHandle, dir, csiDriverName(driverName))
+       
+
        // persist volume info data for teardown
        node := string(p.host.GetNodeName())

```

Then run `hack/update-vendor.sh` to update the vendor. :) @moolen Would you like to have a try?

Reply:
Just compile it, the error:
```
+ go build -o /root/codes/src/github.com/kubeedge/kubeedge/_output/local/bin/edgecore -gcflags= -ldflags '-s -w -buildid= -X github.com/kubeedge/kubeedge/pkg/version.buildDate=2022-02-18T09:35:08Z -X github.com/kubeedge/kubeedge/pkg/version.gitCommit=dd12ab16671269436c5af705316ef074535d7453 -X github.com/kubeedge/kubeedge/pkg/version.gitTreeState=dirty -X github.com/kubeedge/kubeedge/pkg/version.gitVersion=v1.9.0-168+dd12ab16671269-dirty -X github.com/kubeedge/kubeedge/pkg/version.gitMajor=1 -X github.com/kubeedge/kubeedge/pkg/version.gitMinor=9+' github.com/kubeedge/kubeedge/edge/cmd/edgecore
# github.com/kubeedge/kubeedge/edge/pkg/edged/volume/csi
edge/pkg/edged/volume/csi/csi_plugin.go:419:78: cannot use csiDriverName(driverName) (type csiDriverName) as type "k8s.io/kubernetes/pkg/volume/csi".csiDriverName in argument to "k8s.io/kubernetes/pkg/volume/csi".NewMetricsCsi

```

Reply:
yes, that's exactly the error i'm seeing.